arangod/Aql/IResearchViewNode.cpp

```diff
@@ -450,6 +450,46 @@ bool parseOptions(aql::QueryContext& query, LogicalView const& view, aql::AstNod
 // --SECTION--                                                     other helpers
 // -----------------------------------------------------------------------------
 
+bool isSetterVolatile(aql::ExecutionPlan const& plan, 
+                      arangodb::aql::ExecutionNode& setter) {
+  if (!setter.isDeterministic()) {
```

Comment:
I know that I wrote the very first version of this, but I'm wondering now why can't we just use `setter.getLoop()`? :-)

Reply:
Indeed. 👍  Using just getLoop should do the trick. I`ll check it when tests are ready. But I`m sure this will work.